### PR TITLE
Add note about aggregations on floats

### PR DESCRIPTION
**Pull request by suneet-s**
Floating point math is known to be unstable. Due to the way aggregators work
across segments it's possible for the same query operating on the same data to
produce slightly different results.

The same problem exists with any aggregators that are not commutative since
the merge order across segments is not guaranteed.

### CODE REVIEW

File: docs/querying/sql.md [hunk 1]
```diff
@@ -232,6 +232,10 @@ possible for two aggregators in the same SQL query to have different filters.
 
 Only the COUNT aggregation can accept DISTINCT.
 
+> The order of aggregation operations across segments is not deterministic. This means that non-commutative aggregation
+> functions can produce inconsistent results across the same query. Any functions that operate on an input type of "float"
+> may also see these differences in aggregation results across multiple query runs.
```
Comment (gianm) [inline docs/querying/sql.md:237]: Similar comment.

File: docs/development/extensions-core/stats.md [hunk 2]
```diff
@@ -44,6 +44,13 @@ This algorithm was proven to be numerically stable by J.L. Barlow in
 "Error analysis of a pairwise summation algorithm to compute sample variance"
 Numer. Math, 58 (1991) pp. 583--590
 
+> As with all [aggregators](../../querying/sql.md#aggregation-functions), the order of operations across segments is
+> non-deterministic. This means that if this aggregator is used with an input type of "float", the result of the
+> aggregation will not be precisely the same across multiple runs of the query.
+>
+> To produce consistent results, either cast the input type to a "double" or round the variance to
```
Comment (gianm) [inline docs/development/extensions-core/stats.md:51]: Double and float are both floating-point, so should behave the same way in this regard.

File: docs/querying/sql.md [hunk 3]
```diff
@@ -232,6 +232,13 @@ possible for two aggregators in the same SQL query to have different filters.
 
 Only the COUNT aggregation can accept DISTINCT.
 
+> The order of aggregation operations across segments is not deterministic. This means that non-commutative aggregation
+> functions can produce inconsistent results across the same query. 
+>
+> Functions that operates on an input type of "float" or "double" may also see these differences in aggregation
```
Comment (jon-wei) [inline docs/querying/sql.md:238]: operates -> operate

File: docs/development/extensions-core/stats.md [hunk 4]
```diff
@@ -44,6 +44,13 @@ This algorithm was proven to be numerically stable by J.L. Barlow in
 "Error analysis of a pairwise summation algorithm to compute sample variance"
 Numer. Math, 58 (1991) pp. 583--590
 
+> As with all [aggregators](../../querying/sql.md#aggregation-functions), the order of operations across segments is
+> non-deterministic. This means that if this aggregator operates with an input type of "float" or "double", the result
+> of the aggregation will not be precisely the same across multiple runs of the query.
```
Comment (jon-wei) [inline docs/development/extensions-core/stats.md:49]: Suggest "may" instead of "will"

Comment (suneet-s) [reply docs/development/extensions-core/stats.md:49]:
```suggestion
> of the aggregation may not be precisely the same across multiple runs of the query.
```

File: docs/querying/sql.md (hunk 3 above)
Comment (suneet-s) [reply docs/querying/sql.md:238]:
```suggestion
> Functions that operate on an input type of "float" or "double" may also see these differences in aggregation
```

**Pull request merged**